vars/commonPipelineEnvironment.groovy

```diff
@@ -176,16 +176,24 @@ class commonPipelineEnvironment implements Serializable {
         containerProperties.each({key, value ->
             def fileName = ".pipeline/commonPipelineEnvironment/container/${key}"
             if (value && !script.fileExists(fileName)) {
-                //ToDo: check for value type and act accordingly?
-                script.writeFile file: fileName, text: value
+                //ToDo: check for additional value types and act accordingly?
+                if(value instanceof List) {
+                    script.writeFile file: fileName, text: value.join(",")
```

Comment:
what do you think about writing a JSON string in case it is a more complex type (List, Map, ...)

@nevskrem what do you think how we should tackle more complex types?

Reply:
Doesn't the JsonSlurper has a bad reputation in regards of performance? Could we merge the PR and improve it later? This is blocking us from implementing a CI...

Reply:
@OliverNocon, @stiller-leser since we have a rather short lived persistence here it shouldn't impact us too much if we change that later but going for JSON on complex types would be a rather general solution that could be done fairly easy already now. We basically have a required functionality in [JsonUtils](https://github.com/SAP/jenkins-library/blob/master/src/com/sap/piper/JsonUtils.groovy) already, so why not using it immediately?

Reply:
So `JsonUtils.groovyObjectToPrettyJsonString(value)`?

Reply:
@stiller-leser I'd more go for `JsonUtils.groovyObjectToJsonString(...)` since that should be readable enough.